### PR TITLE
Ble backend

### LINKED ISSUE #45: put public key of user in db
generate a public-private key with ed25519 and put the public key of user in db

### PR DESCRIPTION
**Pull request by gngster**
fixes #45 #51

- signature generation and verification implemented on backend 
- added additional button "connect to tag" in login page

### CODE REVIEW

File: backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/ProtectPmLeeFrontendServerApplication.java [hunk 1]
```diff
@@ -22,6 +16,11 @@
 import org.springframework.data.jpa.convert.threeten.Jsr310JpaConverters;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
+import javax.annotation.PostConstruct;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.TimeZone;
+
```
[the remark this answers is not in the document]
Comment (gngster) [reply backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/ProtectPmLeeFrontendServerApplication.java]: Sorry idk eh it just changed i didnt touch but i saw they were still the same >< so i just left it

[the remark this answers is not in the document]
Comment (Zhiyuan-Amos) [reply backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/ProtectPmLeeFrontendServerApplication.java]: Haha it's ok then. Probably IntelliJ just auto-changed it. :)

**Comment by gngster** [comment]
fixes #51

**Comment by gjlam95** [comment]
Login integration with bluetooth looks ok to me. Just a question for @gngster ,will users without BLE be able to log in still?

**Comment by Zhiyuan-Amos** [comment]
Is #50 resolved by this as well? If so, include it in the first comment (scroll all the way up).

**Pull request merged**